packages/components/src/SubHeaderBar/InputTitleSubHeader/InputTitleSubHeader.component.js

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Action } from '../../Actions';
+import theme from './InputTitleSubHeader.scss';
+import { getDefaultTranslate } from '../../translate';
+import Icon from '../../Icon';
+
+function TitleSubHeader({ title, subTitle, onEdit, t }) {
+	return (
+		<div className={classNames(theme['tc-subheader-details-text'], 'tc-subheader-details-text')}>
+			<div
+				className={classNames(
+					theme['tc-subheader-details-text-title'],
+					'tc-subheader-details-text-title',
+				)}
+			>
+				<h1
+					className={classNames(
+						theme['tc-subheader-details-text-title-wording'],
+						'tc-subheader-details-text-title-wording',
+					)}
+				>
+					{title}
+				</h1>
+				{onEdit && (
+					<Action
+						name="action-edit-title"
+						label={t('MODIFY_TOOLTIP', { defaultValue: 'Modify' })}
+						icon="talend-pencil"
+						onClick={onEdit}
+						bsStyle="link"
+						className={classNames(
+							theme['tc-subheader-details-text-title-pencil'],
+							'tc-subheader-details-text-title-pencil',
+						)}
+						hideLabel
+					/>
+				)}
+			</div>
+			{subTitle && (
+				<small
+					className={classNames(
+						theme['tc-subheader-details-text-subtitle'],
+						'tc-subheader-details-text-subtitle',
+					)}
+				>
+					{subTitle}
+				</small>
+			)}
+		</div>
+	);
+}
+
+TitleSubHeader.propTypes = {
+	title: PropTypes.string.isRequired,
+	subTitle: PropTypes.string,
+	onEdit: PropTypes.func,
+	t: PropTypes.func,
+};
+
+TitleSubHeader.defaultProps = {
+	t: getDefaultTranslate,
+};
+
+function selectInput(input) {
+	if (input) {
+		input.select();
+		input.focus();
+	}
+}
+
+function InlineFormSubHeader({ title, inputTextValue, onSubmit, onCancel, onChange, t }) {
+	return (
+		<form className={classNames(theme['tc-subheader-details-form'], 'tc-subheader-details-form')}>
```

Comment:
there is no onSubmit here. so when I play with the storybook it looks like there is a real submit that happens. I guess you should add it